main/test/ca/uwaterloo/flix/library/TestChain.flix

```diff
@@ -1410,6 +1410,22 @@ namespace TestChain {
         Chain.forEach(x -> r := x, List.toChain(1 :: 2 :: 3 :: Nil));
         3 == deref r
 
+    /////////////////////////////////////////////////////////////////////////////
+    // forEachWithIndex                                                        //
+    /////////////////////////////////////////////////////////////////////////////
+
+    @test
+    def forEachWithIndex01(): Bool \ IO =
+        let r = ref 21;
```

Comment:
This style of using a region implicit will soon disappear. So these tests should introduce an explicit region.

Reply:
I've done all the `forEach` and `forEachWithIndex` tests for the stdlib including `forEach` ones for modules like `Option` that weren't in the original PR.